src/tests/fake/references.ts

```diff
@@ -125,3 +126,23 @@ export function mockApiCloneReference(name: string, description: string, referen
         })
         .reply(201, clonedReference);
 }
+
+/**
+ * Sets up a mocked API route for creating an empty reference
+ *
+ * @param name - The name of the reference
+ * @param description - The description of the reference
+ * @param data_type - The data type of the reference
+ * @param organism - The organism of the reference
+ * @returns The nock scope for the mocked API call
+ */
+export function mockApiCreateReference(
+    name: string,
+    description: string,
+    data_type: ReferenceDataType,
+    organism: string,
+) {
+    const reference = createFakeReference({ name, description, data_type, organism });
+
+    return nock("http://localhost").post("/api/refs", { name, description, data_type, organism }).reply(201, reference);
```

Comment:
Does the API return the entire ref? Your typing with the mutation suggests your only getting some value back, but here your returning the entire reference. Update whichever one is incorrect